beacon-chain/state/state-native/setters_consolidation.go

```diff
@@ -23,13 +23,18 @@ func (b *BeaconState) AppendPendingConsolidation(val *ethpb.PendingConsolidation
 	b.lock.Lock()
 	defer b.lock.Unlock()
 
-	b.sharedFieldReferences[types.PendingConsolidations].MinusRef()
-	b.sharedFieldReferences[types.PendingConsolidations] = stateutil.NewRef(1)
-
-	b.pendingConsolidations = append(b.pendingConsolidations, val)
+	pendingConsolidations := b.pendingConsolidations
+	if b.sharedFieldReferences[types.PendingConsolidations].Refs() > 1 {
+		pendingConsolidations = make([]*ethpb.PendingConsolidation, 0, len(b.pendingConsolidations)+1)
+		pendingConsolidations = append(pendingConsolidations, b.pendingConsolidations...)
+		b.sharedFieldReferences[types.PendingConsolidations].MinusRef()
+		b.sharedFieldReferences[types.PendingConsolidations] = stateutil.NewRef(1)
+	}
 
+	b.pendingConsolidations = append(pendingConsolidations, val)
 	b.markFieldAsDirty(types.PendingConsolidations)
-	b.rebuildTrie[types.PendingConsolidations] = true
+	b.addDirtyIndices(types.PendingConsolidations, []uint64{uint64(len(b.pendingConsolidations) - 1)})
```

Comment:
Remove the dirty indices line, this is only used if we maintain a field trie for the particular field